src/PaperG/FirehoundBlob/Facebook/FacebookCreative.php

```diff
@@ -105,18 +126,26 @@ public function toArray()
      */
     public function fromArray($array)
     {
-        $this->type = $this->safeGet($array, self::TYPE);
-        $objects = $this->safeGet($array, self::OBJECTS, []);
-        $objectArray = [];
-        foreach ($objects as $object) {
-            if ($this->type == 'carousel') {
-                $objectArray[] = new FacebookCarouselCreativeData($object);
-            } else {
-                $objectArray[] = new FacebookCreativeData($object); //versioned, might need builder
-            }
+        $this->type       = $this->safeGet($array, self::TYPE);
+        $primary    = $this->safeGet($array, self::PRIMARY);
+        $childAttachments = $this->safeGet($array, self::CHILD_ATTACHMENTS, []);
+
+        $this->primary = $this->createCreativeData($primary);
+
+        $objectsArray = [];
+        foreach ($childAttachments as $object) {
+            $objectsArray[]  = $this->createCreativeData($object);
         }
 
-        $this->objects = $objectArray;
+        $this->childAttachments = $objectsArray;
     }
 
+    private function createCreativeData($creative)
```

Comment:
@a37wong - I've been thinking a bit about this class, and I wonder if I did it wrong.  I think it's kind of odd that a property of the class may contain two different types.  Is the correct way to `extend` FacebookCreative into having a FacebookCarouselCreative, and then just override the variable+method comments?  That seemed like a lot of redundant code, but I'm also not happy with my current implementation.